phpunit.xml.dist

```diff
@@ -1,28 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- https://phpunit.de/manual/current/en/appendixes.configuration.html -->
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
-         colors="true"
-         verbose="true"
+<phpunit colors="true"
+         failOnWarning="false"
          beStrictAboutOutputDuringTests="true"
-         beStrictAboutTodoAnnotatedTests="true"
          beStrictAboutChangesToGlobalState="true"
-         convertErrorsToExceptions="true"
-         convertWarningsToExceptions="true"
-         convertNoticesToExceptions="true"
-         convertDeprecationsToExceptions="true"
-         bootstrap="tests/bootstrap.php"
->
```

Comment:
It's unclear to me why the majority of these settings are being removed.

AFAICS, the only change needed is: `xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.2/phpunit.xsd"` to indicate to PHPUnit the schema used (with 9.2 deliberately chosen as PHPUnit changes some thing in PHPUnit 9.3).

The original schema setting made it dependent on the Composer installed PHPUnit version, which makes the schema setting unstable if PHPUnit 10.0 would also be supported (as PHPUnit 10 made significant changes to the schema).

As for the other values - in part - being set to the default value used by PHPUnit: those defaults have changed for a lot of these across PHPUnit versions, which is why it is important to have these settings explicitly set to make sure tests are being run with the same configuration across PHPUnit versions.

Reply:
I reduced the PHPUnit configuration to be as minimal as possible in order to maximize compatbility between the different supported versions. Fortunately, the schema is not required and can simply be omitted.

Reply:
Do you have a suggestion on how to fix https://github.com/spawnia/phpunit-arraysubset-asserts/actions/runs/4466219397/jobs/7844126983 then?

Reply:
Of course 😉 

I believe this can be fixed by running `phpunit --migrate-configuration` in a separate CI step prior to running the actual tests.
The step should be run conditionally for PHPUnit 10 only.

You can grab the PHPUnit version with something like this:
```yaml
      - name: Grab PHPUnit version
        id: phpunit_version
        run: echo "VERSION=$(vendor/bin/phpunit --version | grep --only-matching --max-count=1 --extended-regexp '\b[0-9]+\.[0-9]+')" >> $GITHUB_OUTPUT
```

... which allows you to check the version in a conditional step after that like so:
```yaml
        if: ${{ steps.phpunit_version.outputs.VERSION >= '10.0' }}
```

As a side-note: I find it weird that PHPUnit would exit with a non-zero exit code when it is only a warning from PHPUnit 10 itself and there are no test failures.
Might be worth opening an issue in the PHPUnit repo about this if there isn't one open already) as this is undesired and unexpected behaviour.

Reply:
I could not get your suggestion to work, but found an alternative.

Reply:
I've been doing some testing with the `--migrate-configuration` option and have found that unfortunately it does not sufficiently convert the configuration.

Most notably, the PHPUnit 10.x `displayDetailsOn*` settings do not get set, while for a tool like this, which is used in the test pipeline by other projects, it is important to have access to information about deprecations in new PHP versions early.

For the PHPUnit Polyfills, I've now solved this by having a separate config file for PHPUnit 10.
If you'd like to see the implementation as an example, please have a look at: https://github.com/Yoast/PHPUnit-Polyfills/pull/111

Reply:
Just noticed that in the upcoming PHPUnit 10.1.0 release there will be new `failOnWarning`, `failOnNotice` and `failOnDeprecation` attributes. Those seem even better/more appropriate.

See:
* sebastianbergmann/phpunit#5196
* https://github.com/sebastianbergmann/phpunit/commit/fb6673f06ef90b43667b7187722e2c840dcdb573

Reply:
We should tackle 10.1 in another PR and continue this discussion there.
WDYT?